### ISSUE TITLE
[JENKINS-63146] Pipeline Clone fails on zOS with SSH Key-pair with Passphrase

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/git-client-plugin#1623]
<p>When attempting to clone a repository using an SSH key-pair in a pipeline, the clone fails with<br/>
Permission denied (publickey).<br/>
if the key has a passphrase. The authentication succeeds if there is no passphrase. Various encodings have been attempted to get past this, including utf8, ISO8859-1, and IBM-1047.</p>

<p>We are using an SSH Agent as a workaround for the time being, but that requires manually modifying the git configuration in new workspaces.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rsbeckerca.jpg" title="rsbeckerca's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rsbeckerca">rsbeckerca</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146" target="_blank">Pipeline Clone fails on zOS with SSH Key-pair with Passphrase</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite">markewaite</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Blocker
<li><b>component(s)</b>: git-client-plugin
<li><b>label(s)</b>: ssh
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2020-08-08T23:01:36+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 2
<li><b>imported</b>: 20251211-071809
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>When attempting to clone a repository using an SSH key-pair in a pipeline, the clone fails with
Permission denied (publickey).
if the key has a passphrase. The authentication succeeds if there is no passphrase. Various encodings have been attempted to get past this, including utf8, ISO8859-1, and IBM-1047.</p>

<p>We are using an SSH Agent as a workaround for the time being, but that requires manually modifying the git configuration in new workspaces.</p></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
Jenkins:2.222.3<br/>
git-client: 3.3.1<br/>
git-plugin: 4.3.0<br/>
z/OS R2.4<br/>
git 2.14
```
</details>

**Comment by jenkins-infra-bot** [comment]
- _Remote links associated with this issue:_


  - [PR-585 - set character set based on environment variable](https://github.com/jenkinsci/git-client-plugin/pull/585)
  - [PR-584 - set passphrase character set based on ibm.system.encoding property](https://github.com/jenkinsci/git-client-plugin/pull/584)

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=394554&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-394554">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thanks for the report and good luck with your investigation.  I don't have access to z/OS so can only offer to code review proposed changes.  Refer to the CredentialsTest in git client plugin if you'd like an automated test environment that can check passphrase protected private keys.</p>
</pre>
</details>

</ul>
</details>

<p>Thanks for the report and good luck with your investigation.  I don't have access to z/OS so can only offer to code review proposed changes.  Refer to the CredentialsTest in git client plugin if you'd like an automated test environment that can check passphrase protected private keys.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rsbeckerca.jpg" title="rsbeckerca's avatar" /> rsbeckerca:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=394558&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-394558">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thanks. Any pointers on where the passphrase is passed in? I'm guessing this is an encoding issue. The credentials comes from a controller on x86 Gentoo Docker image of Jenkins, going to a z/OS agent. I'm not at all sure what happens with all the <b>if (isZos()) {</b> code branches. Sadly, this is not my native platform either, but I do have a machine on which I can try to debug the agent (with difficulty).</p>
</pre>
</details>

</ul>
</details>

<p>Thanks. Any pointers on where the passphrase is passed in? I'm guessing this is an encoding issue. The credentials comes from a controller on x86 Gentoo Docker image of Jenkins, going to a z/OS agent. I'm not at all sure what happens with all the <b>if (isZos()) {</b> code branches. Sadly, this is not my native platform either, but I do have a machine on which I can try to debug the agent (with difficulty).</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=394559&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-394559">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Search for the 'passphrase' in git-client-plugin.  <a href="https://github.com/jenkinsci/git-client-plugin/blob/eeec334af0b6447f3db9fb88d55728911a092d73/src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java#L1962" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-client-plugin/blob/eeec334af0b6447f3db9fb88d55728911a092d73/src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java#L1962</a></p>
</pre>
</details>

</ul>
</details>

<p>Search for the 'passphrase' in git-client-plugin.  <a href="https://github.com/jenkinsci/git-client-plugin/blob/eeec334af0b6447f3db9fb88d55728911a092d73/src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java#L1962" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-client-plugin/blob/eeec334af0b6447f3db9fb88d55728911a092d73/src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java#L1962</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rsbeckerca.jpg" title="rsbeckerca's avatar" /> rsbeckerca:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=394563&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-394563">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I am not sure about the line at 2167:</p>
<div class='table-wrap'>
<table class='confluenceTable'><tbody>
<tr>
<td class='confluenceTd'><tt>try (PrintWriter w = new PrintWriter(passphraseFile, "UTF-8")) {</tt></td>
</tr>
</tbody></table>
</div>


<div class='table-wrap'>
<table class='confluenceTable'><tbody>
<tr>
<td class='confluenceTd'><tt>   w.println(Secret.toString(sshUser.getPassphrase()));</tt></td>
</tr>
</tbody></table>
</div>


<p><tt>}</tt></p>

<p>On the zOS machine, I think the passphrase file needs to be encoded in IBM-1047. I will verify with my contacts at IBM on this.</p>
</pre>
</details>

</ul>
</details>

<p>I am not sure about the line at 2167:</p>
<div class='table-wrap'>
<table class='confluenceTable'><tbody>
<tr>
<td class='confluenceTd'><tt>try (PrintWriter w = new PrintWriter(passphraseFile, "UTF-8")) {</tt></td>
</tr>
</tbody></table>
</div>


<div class='table-wrap'>
<table class='confluenceTable'><tbody>
<tr>
<td class='confluenceTd'><tt>   w.println(Secret.toString(sshUser.getPassphrase()));</tt></td>
</tr>
</tbody></table>
</div>


<p><tt>}</tt></p>

<p>On the zOS machine, I think the passphrase file needs to be encoded in IBM-1047. I will verify with my contacts at IBM on this.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rsbeckerca.jpg" title="rsbeckerca's avatar" /> rsbeckerca:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=394597&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-394597">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>So far, it looks like the IBM team is requesting a parameter to set the passphrase encoding rather than hard coding "UTF-8". Would that be possible?</p>
</pre>
</details>

</ul>
</details>

<p>So far, it looks like the IBM team is requesting a parameter to set the passphrase encoding rather than hard coding "UTF-8". Would that be possible?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=394600&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-394600">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Is it the same encoding used for the process on the agent or does it need to be independent of the encoding used for the agent?</p>

<p>I could envision an agent specific environment variable defined for the zOS agent and not defined on other agents.  If that variable is defined, then it could be used when encoding those temporary files that transmit credentials information to command line git.</p>

<p>If your repositories are not huge, you might consider enabling JGit and using JGit instead of command line git.  It does not use temporary files for credentials so it should not rely on file encoding.</p>
</pre>
</details>

</ul>
</details>

<p>Is it the same encoding used for the process on the agent or does it need to be independent of the encoding used for the agent?</p>

<p>I could envision an agent specific environment variable defined for the zOS agent and not defined on other agents.  If that variable is defined, then it could be used when encoding those temporary files that transmit credentials information to command line git.</p>

<p>If your repositories are not huge, you might consider enabling JGit and using JGit instead of command line git.  It does not use temporary files for credentials so it should not rely on file encoding.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rsbeckerca.jpg" title="rsbeckerca's avatar" /> rsbeckerca:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=394601&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-394601">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The situation is that original passphrase is coming from a UTF8 controller going to an IBM-1047 agent. So yes, this should be something on the agent config. Maybe something like -Dibm.passphrase.encoding=IBM-1047. I already tried JGit - it does not seem to like being on the platform at all. I could not make it work on R2.3 at all - either standalone or inside Jenkins.</p>
</pre>
</details>

</ul>
</details>

<p>The situation is that original passphrase is coming from a UTF8 controller going to an IBM-1047 agent. So yes, this should be something on the agent config. Maybe something like -Dibm.passphrase.encoding=IBM-1047. I already tried JGit - it does not seem to like being on the platform at all. I could not make it work on R2.3 at all - either standalone or inside Jenkins.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=394602&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-394602">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I'd prefer a name like <tt>GitSCM.credentials.file.encoding</tt> since there may be other platforms that want to change the encoding of the files.</p>

<p>Can the encoding be used for all the git credentials files (username/password, private key, and passphrase files)?</p>
</pre>
</details>

</ul>
</details>

<p>I'd prefer a name like <tt>GitSCM.credentials.file.encoding</tt> since there may be other platforms that want to change the encoding of the files.</p>

<p>Can the encoding be used for all the git credentials files (username/password, private key, and passphrase files)?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rsbeckerca.jpg" title="rsbeckerca's avatar" /> rsbeckerca:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=394604&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-394604">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>That name sounds ok. The thing is, it looks like the private key is going across fine, it's just the passphrase that is not working because it's UTF-8. As above, when there is no passphrase, GitSCM authenticates fine and does the clone correctly. It seems to be just the passphrase. Can you MacGyver a version for me with a candidate and I will try it?</p>
</pre>
</details>

</ul>
</details>

<p>That name sounds ok. The thing is, it looks like the private key is going across fine, it's just the passphrase that is not working because it's UTF-8. As above, when there is no passphrase, GitSCM authenticates fine and does the clone correctly. It seems to be just the passphrase. Can you MacGyver a version for me with a candidate and I will try it?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=394605&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-394605">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Not until after work today (6 or 8 hours from now).  Git plugin maintenance is a personal time thing.</p>
</pre>
</details>

</ul>
</details>

<p>Not until after work today (6 or 8 hours from now).  Git plugin maintenance is a personal time thing.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rsbeckerca.jpg" title="rsbeckerca's avatar" /> rsbeckerca:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=394606&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-394606">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>At your convenience <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</ul>
</details>

<p>At your convenience <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rsbeckerca.jpg" title="rsbeckerca's avatar" /> rsbeckerca:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=394717&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-394717">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I had confirmation of what's going on. On z/OS, files in the USS space (used by Jenkins) there is a tag attribute of each file that describes the encoding of the file as these can be a mixed bag. The default encoding on the box is assumed to be IBM-1047, but may not necessarily be so. The pipeline code seems to be interpreted correctly because the default encoding of the agent was set to ISO8859-1, so the shell steps understand that and pass the contents appropriately. The passphrase file, being hard-coded to UTF-8 and <b>untagged</b> is interpreted by git as IBM-1047, which it is not, so the authentication fails. So looks like this is just the passphrase file, not all credentials. I put in an interception script to change the tag of the passphrase file to ISO8859-1 and that got the authentication to work - a hack not a generic solution. So I think we're on the right track by allowing the passphrase file to be encoded as something other than UTF-8.</p>
</pre>
</details>

</ul>
</details>

<p>I had confirmation of what's going on. On z/OS, files in the USS space (used by Jenkins) there is a tag attribute of each file that describes the encoding of the file as these can be a mixed bag. The default encoding on the box is assumed to be IBM-1047, but may not necessarily be so. The pipeline code seems to be interpreted correctly because the default encoding of the agent was set to ISO8859-1, so the shell steps understand that and pass the contents appropriately. The passphrase file, being hard-coded to UTF-8 and <b>untagged</b> is interpreted by git as IBM-1047, which it is not, so the authentication fails. So looks like this is just the passphrase file, not all credentials. I put in an interception script to change the tag of the passphrase file to ISO8859-1 and that got the authentication to work - a hack not a generic solution. So I think we're on the right track by allowing the passphrase file to be encoded as something other than UTF-8.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=394798&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-394798">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I've submitted two pull requests that try to provide a character set adjustment for the zOS environment without risking breaking other use cases.</p>

<p>The incremental build with the ibm encoding property is  <span class="error">&#91;^git-client-3.3.2-PR-584-ibm.encoding.property.hpi&#93;</span> </p>

<p>The incremental build with the GitSCM.credentials encoding environment variable is  <span class="error">&#91;^git-client-3.3.2-PR-585-credentials-file.encoding.hpi&#93;</span> </p>
</pre>
</details>

</ul>
</details>

<p>I've submitted two pull requests that try to provide a character set adjustment for the zOS environment without risking breaking other use cases.</p>

<p>The incremental build with the ibm encoding property is  <span class="error">&#91;^git-client-3.3.2-PR-584-ibm.encoding.property.hpi&#93;</span> </p>

<p>The incremental build with the GitSCM.credentials encoding environment variable is  <span class="error">&#91;^git-client-3.3.2-PR-585-credentials-file.encoding.hpi&#93;</span> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rsbeckerca.jpg" title="rsbeckerca's avatar" /> rsbeckerca:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=394822&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-394822">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Environment variables cannot have this particular format, as in GitSCM.credentials.file.encoding, at least not on z/OS. I can use -DGitSCM.credentials.file.encoding or a different name, but this combination doesn't work. Can you change this to System.getProperty("GitSCM.credentials.file.encoding") instead of System.getenv("GitSCM.credentials.file.encoding"), please?</p>
</pre>
</details>

</ul>
</details>

<p>Environment variables cannot have this particular format, as in GitSCM.credentials.file.encoding, at least not on z/OS. I can use -DGitSCM.credentials.file.encoding or a different name, but this combination doesn't work. Can you change this to System.getProperty("GitSCM.credentials.file.encoding") instead of System.getenv("GitSCM.credentials.file.encoding"), please?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=394824&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-394824">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Sure, it can use a property instead of an environment variable.  New build will be available on the CI job for that PR.</p>

<p>In order to make the property match with other properties that control the command line git implementation, the name of the property is now:</p>

<p><tt>org.jenkinsci.plugins.gitclient.CliGitAPIImpl.credentials.file.encoding</tt></p>
</pre>
</details>

</ul>
</details>

<p>Sure, it can use a property instead of an environment variable.  New build will be available on the CI job for that PR.</p>

<p>In order to make the property match with other properties that control the command line git implementation, the name of the property is now:</p>

<p><tt>org.jenkinsci.plugins.gitclient.CliGitAPIImpl.credentials.file.encoding</tt></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rsbeckerca.jpg" title="rsbeckerca's avatar" /> rsbeckerca:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=394854&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-394854">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>CI did not build <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/sad.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</ul>
</details>

<p>CI did not build <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/sad.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rsbeckerca.jpg" title="rsbeckerca's avatar" /> rsbeckerca:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=394899&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-394899">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>It does not look like the build failure was related to the PR; rather, it looks like the CI job has an environment issue.</p>
</pre>
</details>

</ul>
</details>

<p>It does not look like the build failure was related to the PR; rather, it looks like the CI job has an environment issue.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rsbeckerca.jpg" title="rsbeckerca's avatar" /> rsbeckerca:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=394945&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-394945">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Currently under test on my z/OS system</p>
</pre>
</details>

</ul>
</details>

<p>Currently under test on my z/OS system</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rsbeckerca.jpg" title="rsbeckerca's avatar" /> rsbeckerca:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=394947&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-394947">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Looks good. I ran two different jobs - one on a new workspace, one on an existing workspace with no updates. The expected messages came out correctly (repository name changed to protect the guilty):</p>

<p><tt>&gt; git --version # timeout=10</tt>
 using GIT_SSH to set credentials
 <tt>Using passphrase charset 'IBM1047'</tt>
 <tt>&gt; git fetch --tags --progress – git@Bitbucket.org:REPO-SLUG.git +refs/heads/<b>:refs/remotes/origin/</b> # timeout=10</tt>
 <tt>Checking out Revision ce210a4618766170a0dd9cc6f2aee4a8563d4070 (refs/remotes/origin/development)</tt></p>

<p>and we had no authentication issues.</p>

<p>As a side nit - the plugin does not recognize the repository format - probably also an encoding issue but that would be a separate case.</p>

<p>My sincere thanks.</p>
</pre>
</details>

</ul>
</details>

<p>Looks good. I ran two different jobs - one on a new workspace, one on an existing workspace with no updates. The expected messages came out correctly (repository name changed to protect the guilty):</p>

<p><tt>> git --version # timeout=10</tt><br/>
 using GIT_SSH to set credentials<br/>
 <tt>Using passphrase charset 'IBM1047'</tt><br/>
 <tt>> git fetch --tags --progress – git@​Bitbucket.org:REPO-SLUG.git +refs/heads/<b>:refs/remotes/origin/</b> # timeout=10</tt><br/>
 <tt>Checking out Revision ce210a4618766170a0dd9cc6f2aee4a8563d4070 (refs/remotes/origin/development)</tt></p>

<p>and we had no authentication issues.</p>

<p>As a side nit - the plugin does not recognize the repository format - probably also an encoding issue but that would be a separate case.</p>

<p>My sincere thanks.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=395182&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-395182">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rsbeckerca" class="user-hover" rel="rsbeckerca">rsbeckerca</a> I'm unassigning this issue from me now.  I won't merge the proposed pull request until you can confirm that the entire use case is working.  Keep this issue updated as you make further progress.  Thanks!</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rsbeckerca" class="user-hover" rel="rsbeckerca">rsbeckerca</a> I'm unassigning this issue from me now.  I won't merge the proposed pull request until you can confirm that the entire use case is working.  Keep this issue updated as you make further progress.  Thanks!</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rsbeckerca.jpg" title="rsbeckerca's avatar" /> rsbeckerca:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=395183&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-395183">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> We have been using the fix for a week with no issues. I can confirm that the SSH key-pair is correctly authenticating with this option. The repository format recognition is a completely separate issue and unrelated. So the entire use case is working.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> We have been using the fix for a week with no issues. I can confirm that the SSH key-pair is correctly authenticating with this option. The repository format recognition is a completely separate issue and unrelated. So the entire use case is working.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=395185&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-395185">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thanks.  Then all I need to do is add the documentation that describes the property.  That's a good excuse to document the other properties provided by the plugin as well.</p>
</pre>
</details>

</ul>
</details>

<p>Thanks.  Then all I need to do is add the documentation that describes the property.  That's a good excuse to document the other properties provided by the plugin as well.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=395209&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-395209">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rsbeckerca" class="user-hover" rel="rsbeckerca">rsbeckerca</a> I've added documentation for the <a href="https://github.com/MarkEWaite/git-client-plugin/blob/add-property-docs/README.adoc#plugin-properties" class="external-link" target="_blank" rel="nofollow noopener">other properties</a>.  Does the layout look reasonable to you?  If so, then I'll submit a pull request for that documentation, then once it is merged, I'll update the docs in the PR for this property.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rsbeckerca" class="user-hover" rel="rsbeckerca">rsbeckerca</a> I've added documentation for the <a href="https://github.com/MarkEWaite/git-client-plugin/blob/add-property-docs/README.adoc#plugin-properties" class="external-link" target="_blank" rel="nofollow noopener">other properties</a>.  Does the layout look reasonable to you?  If so, then I'll submit a pull request for that documentation, then once it is merged, I'll update the docs in the PR for this property.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rsbeckerca.jpg" title="rsbeckerca's avatar" /> rsbeckerca:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=395211&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-395211">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I like the format. Actually learned something about tags that I was annoying - I want my tags being updated silently as long as they are verified. Side issue. Go for PR <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</ul>
</details>

<p>I like the format. Actually learned something about tags that I was annoying - I want my tags being updated silently as long as they are verified. Side issue. Go for PR <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-63146?focusedId=395731&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-395731">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Released in git client plugin 3.4.0 with the properties documented in <a href="https://plugins.jenkins.io/git-client/#plugin-properties" class="external-link" target="_blank" rel="nofollow noopener">plugin properties</a></p>
</pre>
</details>

</ul>
</details>

<p>Released in git client plugin 3.4.0 with the properties documented in <a href="https://plugins.jenkins.io/git-client/#plugin-properties" class="external-link" target="_blank" rel="nofollow noopener">plugin properties</a></p>
